### DISCUSSION TITLE
How does `straight.el` deal with recipes from melpa?

### DISCUSSION THREAD

**Discussion by eppolito** [raxod502/straight.el#984] [Q&A]
I recently tried to `(straight-use-package 'dirvish)` and was surprised to get an error that the package does not exist.  I found the recipe on `melpa` and it seems fine, but I get an error that the recipe is malformed (using `emacs -Q -l ~/minimal-straight-setup-from-docs.el`).

I'm wondering what's going on here?  I noticed that the [recipe for dirvish](https://github.com/melpa/melpa/blob/master/recipes/dirvish) is a bit more complicated than others I've seen (e.g. [counsel](https://github.com/melpa/melpa/blob/master/recipes/counsel)), as it involves a wildcard.

PS. I did figure it out in the end, using a modified version of the recipe on `melpa`:
`(straight-use-package '(dirvish :type git :repo "alexluigit/dirvish" :host github))`

**Comment by progfolio** [comment]
> I recently tried to (straight-use-package 'dirvish) and was surprised to get an error that the package does not exist.

What was the error? Please always include the error's text in future reports.

Everything seems to be working on my end. Perhaps your MELPA recipe repository needed to be updated? We have the `straight-pull-recipe-repositories` command if that was the case.



> I did figure it out in the end, using a modified version of the recipe on melpa

The modified recipe you've provided will not copy over the files in dirvish's "extensions" directory on its own. However, the default value of `straight-allow-recipe-inheritance` is `t`, so your recipe is likely inheriting from the MELPA recipe.
Note the difference in the following test cases, where I explicitly disable inheritance for the recipe you've provided:

[Yodel](https://github.com/progfolio/yodel) Report (2022-08-05 14:54:36):

```emacs-lisp
(yodel
  :user-dir "melpa-recipe"
  :packages*
  dirvish
  :formatter yodel-format-as-github-markdown
  :post*
  (let ((default-directory (straight--build-dir "dirvish")))
    (message "dirvish build: %s " (straight--process-output "bash" "-c" "ls *.el"))))
```

<details><summary>STDOUT:</summary>

```emacs-lisp
Loading /tmp/melpa-recipe/straight-bootstrap-snippet.el (source)...
Bootstrapping straight.el...
Bootstrapping straight.el...done
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done
Cloning dirvish...
Cloning dirvish...done
Building dirvish...
Building dirvish → Cloning transient...
Building dirvish → Cloning transient...done
Building dirvish → Building transient...
Building dirvish → Building transient → Cloning compat...
Building dirvish → Building transient → Cloning compat...done
Building dirvish → Building transient → Building compat...
Building dirvish → Building transient → Building compat...done
Building dirvish → Building transient...
Building dirvish → Building transient...done
Building dirvish...
Building dirvish...done

dirvish build:
dirvish-autoloads.el
dirvish-collapse.el
dirvish.el
dirvish-emerge.el
dirvish-extras.el
dirvish-fd.el
dirvish-history.el
dirvish-icons.el
dirvish-layout.el
dirvish-ls.el
dirvish-media.el
dirvish-menu.el
dirvish-narrow.el
dirvish-peek.el
dirvish-quick-access.el
dirvish-side.el
dirvish-subtree.el
dirvish-vc.el
dirvish-yank.el
```

</details>
<details><summary>Environment</summary>

- **emacs version**: GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.34, cairo version 1.17.6)
 of 2022-08-02
- **system type**: gnu/linux

<details open><summary>Packages</summary>

| Name                                               | Branch | Commit                                                                                   | Date       | Source          |
|----------------------------------------------------|--------|------------------------------------------------------------------------------------------|------------|-----------------|
| [dirvish](https://github.com/alexluigit/dirvish)   | main   | https://github.com/alexluigit/dirvish/commit/8f1b38d2b8b80bf0adc721d4eb04cdad2b140ac9    | 2022-08-05 | melpa           |
| [transient](https://github.com/magit/transient)    | master | https://github.com/magit/transient/commit/389d2bffff1d7ada561688861ca67b7a360cf352       | 2022-08-03 | melpa           |
| [compat](https://github.com/emacs-straight/compat) | master | https://github.com/emacs-straight/compat/commit/2a9cf8b7bdb5bf2cf4ee73b2c586ec9d736cb4f6 | 2022-07-30 | gnu-elpa-mirror |

</details>

</details>

[Yodel](https://github.com/progfolio/yodel) Report (2022-08-05 14:54:41):

```emacs-lisp
(yodel
  :user-dir "modified-recipe"
  :packages* '(dirvish :type git :repo "alexluigit/dirvish" :host github :inherit nil)
  :formatter yodel-format-as-github-markdown
  :post*
  (let ((default-directory (straight--build-dir "dirvish")))
    (message "dirvish build: %s " (straight--process-output "bash" "-c" "ls *.el"))))
```

<details><summary>STDOUT:</summary>

```emacs-lisp
Loading /tmp/modified-recipe/straight-bootstrap-snippet.el (source)...
Bootstrapping straight.el...
Bootstrapping straight.el...done
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...
Looking for emacsmirror-mirror recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done
Cloning dirvish...
Cloning dirvish...done
Building dirvish...
Building dirvish → Cloning transient...
Building dirvish → Cloning transient...done
Building dirvish → Building transient...
Building dirvish → Building transient → Cloning compat...
Building dirvish → Building transient → Cloning compat...done
Building dirvish → Building transient → Building compat...
Building dirvish → Building transient → Building compat...done
Building dirvish → Building transient...
Building dirvish → Building transient...done
Building dirvish...
Building dirvish...done

dirvish build:
dirvish-autoloads.el
dirvish.el
```

</details>
<details><summary>Environment</summary>

- **emacs version**: GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.34, cairo version 1.17.6)
of 2022-08-02
- **system type**: gnu/linux

<details open><summary>Packages</summary>

| Name                                               | Branch | Commit                                                                                   | Date       | Source          |
|----------------------------------------------------|--------|------------------------------------------------------------------------------------------|------------|-----------------|
| [dirvish](https://github.com/alexluigit/dirvish)   | main   | https://github.com/alexluigit/dirvish/commit/8f1b38d2b8b80bf0adc721d4eb04cdad2b140ac9    | 2022-08-05 | nil             |
| [transient](https://github.com/magit/transient)    | master | https://github.com/magit/transient/commit/389d2bffff1d7ada561688861ca67b7a360cf352       | 2022-08-03 | melpa           |
| [compat](https://github.com/emacs-straight/compat) | master | https://github.com/emacs-straight/compat/commit/2a9cf8b7bdb5bf2cf4ee73b2c586ec9d736cb4f6 | 2022-07-30 | gnu-elpa-mirror |

</details>

</details>
